### PR TITLE
Finished Models, Repositories and UseCases

### PR DESCRIPTION
**Pull request by RafaX8**
## Description de la tarea

Started with domain layer of the project. Finished first take on models, repositories and UseCases.

## ¿Cómo se ha implementado?

Created files in domain package to define domain models of data (only Movie --> need to know if more objects will be included).

Created a repository for domain models:
- Movie --> getMovies() ... and getMovieById(...)...

Created UseCases for repositories Functions:
- Movie --> FeedUseCase and DetailUseCase

## Keywords

Domain, Models, Repository, UseCase.

### CODE REVIEW

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            movie.rating,
+            movie.year,
+            movie.poster
+            )
+
+    }
+
+}
+
+data class MovieDetail (
```
Comment (jmperezra) [inline app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt]: MovieDetail is  business model??

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                it.genre,
+                it.rating,
+                it.poster
+            )
+        }
+    }
+
+}
+
+data class MovieFeed (
```
Comment (jmperezra) [inline app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt]: MovieFeed is an application model, please, you should move it inside of use case

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/Repositories.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.rafaelmardom.movies_and_shows_api.domain
+
+interface MovieRepository {
+    fun getMovies(): List<Movie>
```
Comment (jmperezra) [inline app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/Repositories.kt]: the name of method: getMoviees or getMovieById repeat 'movie'. You can use:
getAll() and getById or findById...

**Review by jmperezra — CHANGES_REQUESTED** [review]
Please, check the comments.

Thanks!

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt (hunk 1 above)
Comment (RafaX8) [reply app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt]: I think I don't understand the question, but I've moved the data class inside of UseCase the same way I've done with MovieFeed below.

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt (hunk 2 above)
Comment (RafaX8) [reply app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt]: Done!

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/Repositories.kt (hunk 3 above)
Comment (RafaX8) [reply app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/Repositories.kt]: Done!

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        return MovieDetail(
+            movie.id,
+            movie.title,
+            movie.plot,
+            movie.genre,
+            movie.rating,
+            movie.year,
+            movie.poster
+            )
+
```
Comment (jmperezra) [inline app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMovieDetailUseCase.kt]: extraline, please, remove it.

File: app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package com.rafaelmardom.movies_and_shows_api.domain
+
+class GetMoviesFeedUseCase (private val movieRepository: MovieRepository) {
+
+    fun execute () : List<MovieFeed> {
+        val movies = movieRepository.getAll()
```
Comment (jmperezra) [inline app/src/main/java/com/rafaelmardom/movies_and_shows_api/domain/GetMoviesFeedUseCase.kt]: Puedes quitar la variable movies, quedaría así:
``
movieRepository.getAll().map{
  ...
}
``

**Review by jmperezra — APPROVED** [review]
La PR está ok, revisa los comentarios please.

**Pull request merged**